### PR TITLE
Fix/master/3.0.1

### PR DESCRIPTION
**Pull request by anestos**
- Bump the SDK version to 3.0.1
- Add all incoming attributes from https://api.github.com/user to the subjectAttributes
- Fix webServiceClient calls by using withPath()

### CODE REVIEW

File: src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java [hunk 1]
```diff
@@ -148,7 +133,8 @@ public CallbackGetRequestModel preProcess(Request request, Response response)
             throw _exceptionFactory.internalServerException(ErrorCode.EXTERNAL_SERVICE_ERROR);
         }
 
-        HttpResponse userInfoResponse = getWebServiceClient("https://api.github.com/user")
+        HttpResponse userInfoResponse = getWebServiceClient("https://api.github.com/")
+                .withPath("user")
```
Comment (travisspencer) [inline src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java]: I wonder if this was broken as a part of IS-2495.

File: src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java [hunk 2]
```diff
@@ -204,7 +190,8 @@ private WebServiceClient getWebServiceClient(String uri)
 
     private Map<String, Object> redeemCodeForTokens(CallbackGetRequestModel requestModel)
     {
-        HttpResponse tokenResponse = getWebServiceClient("https://github.com/login/oauth/access_token")
+        HttpResponse tokenResponse = getWebServiceClient("https://github.com/")
+                .withPath("login/oauth/access_token")
```
Comment (travisspencer) [inline src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java]: I wonder how many plug-ins where broken by this accidental change?

**Review by travisspencer — APPROVED** [review]

File: src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java (hunk 2 above)
Comment (travisspencer) [reply src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java]: This test is passing locally here:
```groovy
def "Can create a request that uses the path provided in the URI create overload"() {
        given:
        def path = "/users"
        URI expectedUri = URI.create("https://github.com" + path)
        WebServiceClientFactory factory = new DefaultWebServiceClientFactory()

        when:
        def actualUri = factory.create(expectedUri).request().get().uri()

        then:
        expectedUri == actualUri
        path == actualUri.path // Should be true if above is true, but URI does funny stuff from time to time so be sure
}
```
Why was this change needed?

Comment (anestos) [reply src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java]: The problem was not there, but how we instantiate the WebServiceClient in line 181.
`return _webServiceClientFactory.create(h).withHost(u.getHost());`
the path is not considered. I ll fix that instead

File: src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java [hunk 3]
```diff
@@ -180,7 +177,7 @@ private WebServiceClient getWebServiceClient(String uri)
                                 requiredScheme, configuredScheme));
             }
 
-            return _webServiceClientFactory.create(h).withHost(u.getHost());
+            return _webServiceClientFactory.create(h).withHost(u.getHost()).withPath(u.getPath());
```
Comment (travisspencer) [inline src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java:180]: ah, yes. This was a bug in this plug-in. Whew. SDK is fine and no unexpected breaking change :)

**Pull request merged**